03 - Data Science and Custom Skills/Azure Function Skills/SampleSkills/0_CustomSkillTemplate.cs

```diff
@@ -0,0 +1,171 @@
+/****************************************************************************************
+ * 
+ * This template should make it easy for you to create a custom skill as
+ * it takes care of most of the boilerplate code for serialization and  deserialization.
+ * 
+ * STEPS:
+ * 1. Define the input fields for your skill.
+ *    Modify the InputRecordData class.
+ * 
+ * 2. Define the output fields for your skill.
+ *    Modify the OutputRecordData class.
+ *    
+ * 3. Define what action your will will take to enrich/transform the input into the output.
+ *    Modify the DoWorkHere class.
+ *    
+ ****************************************************************************************/
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+
+namespace SampleSkills
+{
+
+    public static class CustomSkillTemplate
+    {
+
+        #region Class used to deserialize the request
+        public class InputRecord
+        {
+            public class InputRecordData
+            {
+                public string myInputField;
+            }
+
+            public string recordId { get; set; }
+            public InputRecordData data { get; set; }
+        }
+
+        private class WebApiRequest
+        {
+            public List<InputRecord> values { get; set; }
+        }
+        #endregion
+
+        #region Classes used to serialize the response
+        public class OutputRecord
+        {
+            public class ComputerVisionObject
+            {
+                [JsonProperty(PropertyName = "object")]
+                public string cvObject { get; set; }
+                public string confidence { get; set; }
+            }
+
+            public class OutputRecordData
+            {
+                public string myOutputField { get; set; }
+            }
+
+            public class OutputRecordErrors
+            {
+                public string message { get; set; }
+            }
+
+            public class OutputRecordWarnings
```

Comment:
Should just be able to reuse the same class for warnings and errors since its the same contract.  Again you can see how it was defined for JFK: https://github.com/Microsoft/AzureSearch_JFK_Files/blob/master/JfkWebApiSkills/JfkWebApiSkills/WebApiSkill/WebAPISkillContract.cs